lib/elixir/src/elixir_parser.yrl

```diff
@@ -713,6 +713,16 @@ build_op({_Kind, Location, 'not in'}, Left, Right) ->
 build_op({_Kind, Location, Op}, Left, Right) ->
   {Op, newlines_op(Location) ++ meta_from_location(Location), [Left, Right]}.
 
+build_unary_op({capture_op, Location, Op}, {_,_,[{_,[{no_parens,true} | _],_}, 0]} = Expr) ->
```

Comment:
We cannot rely on `{no_parens, true}` being the first element in the metadata. Instead you should use `lists:keyfind/3` to look through all elements.

Reply:
Since the capture can be built with macros and not only a literal I think we should check for parenthesis in the expansion pass like you do for the `Foo.bar` syntax.